iree/compiler/Dialect/Util/IR/UtilDialect.cpp

```diff
@@ -143,6 +143,36 @@ void UtilDialect::getCanonicalizationPatterns(
   results.insert<FoldDimOp<tensor::DimOp>>(getContext());
 }
 
+// Since all details of the interface are provided via default implementations,
+// we can just have one templated external model to apply per op, vs one
+// explicit model per op.
+template <typename OpTy>
+struct GenericNumericCastOpInterface
```

Comment:
Interesting. I dont see any implementations of the external model methods. So I am assuming this is just using the default implementation of the interface for all these ops. I hadnt tried it, but this is neat.

Reply:
Yes, it is just coming from the default implementation, which it turns out has access to the template parameter of the applied type. I haven't tracked through all of why it works, but I have confirmed that it does :)